doc/can-query-logic.md

```diff
@@ -14,139 +14,141 @@ data caching and real-time behavior.
-    might not produce the desired results:
-
-    ```js
-    var queryLogic = new QueryLogic({identity: ["id"]});
-    queryLogic.union(
-        {filter: {age: 7}},
-        {filter: {age: "07"}}) //-> {filter: {age: {$in: [7,"07"]}}}
-    ```
-    Use types like [can-data-types/maybe-number/maybe-number] if you want to add basic
-    type coercion:
-
-    ```js
-    var queryLogic = new QueryLogic({
-        identity: ["id"],
-        keys: {age: MaybeNumber}
-    });
-    queryLogic.union(
-        {filter: {age: 7}},
-        {filter: {age: "07"}}) //-> {filter: {age: 7}}
-    ```
-
-    If you need even more special key behavior, read [defining properties with special logic](#Definingfilterpropertieswithspeciallogic).
+  ```
+  @codepen
+  @highlight 6-15,only
+
+@param {function|can-reflect/schema} schemaOrType Defines the behavior of keys on a [can-query-logic/query]. This is done with either:
+
+- A constructor function that supports [can-reflect.getSchema can-reflect.getSchema]. Currently, [can-define/map/map] supports the `can.getSchema` symbol:
+
+  @sourceref ./can-query-logic-models/todo-example.js
+  @codepen
+  @highlight 3,10,only
+
+- A [can-reflect.getSchema schema object] that looks like the following:
+
+  ```js
+  import {QueryLogic, MaybeNumber, MaybeString, MaybeBoolean} from "can";
+
+  const queryLogic = new QueryLogic({
+    // keys that uniquely represent this type
+    identity: ["id"],
+    keys: {
+      id: MaybeNumber,
+      name: MaybeString,
+      complete: MaybeBoolean
+    }
+  });
+
+  const result = queryLogic.filterMembers({ filter: {complete: false}}, [
+    {id: "1", name: "Justin", complete: "truthy"},
+    {id: "2", name: "Paula", complete: ""},
+    {id: "3", name: "Kevin", complete: true}
+  ]);
+
+  console.log( result );
+  ```
+  <!-- Example doesn't work. Issue open: https://github.com/canjs/can-data-types/issues/7 -->
+  <!-- @codepen -->
+
+  Note that if a key type (ex: `name: MaybeString`) is __not__ provided, filtering by that
+  key will still work, but there won't be any type coercion. For example, the following
+  might not produce the desired results:
+
+  ```js
+  import {QueryLogic} from "can";
+
+  const queryLogic = new QueryLogic({identity: ["id"]});
+  const unionized = queryLogic.union(
+    {filter: {age: 7}},
+    {filter: {age: "07"}}
+  );
+  console.log( JSON.stringify( unionized ) ); //-> "{'filter':{'age':{'$in':[7,'07']}}}"
+  ```
+  @codepen
+
+  Use types like [can-data-types/maybe-number/maybe-number] if you want to add basic
+  type coercion:
+
+  ```js
+  import {QueryLogic, MaybeNumber} from "can";
+
+  const queryLogic = new QueryLogic({
+    identity: ["id"],
+    keys: {age: MaybeNumber}
+  });
+  const unionized = queryLogic.union(
+    {filter: {age: 7}},
+    {filter: {age: "07"}}
+  );
+  console.log( JSON.stringify( unionized ) ); //-> {filter: {age: 7}}
+  ```
+  <!-- Example doesn't work. Issue open: https://github.com/canjs/can-data-types/issues/7 -->
```

Comment:
this is working.  it's ok to leave in the codepen if the next release will work.